Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -27,6 +38,83 @@ void RegExpConstructor::initialize(Realm& realm)
+    define_native_accessor(realm, "$8", group_8_getter, {}, Attribute::Configurable);
+    define_native_accessor(realm, "$9", group_9_getter, {}, Attribute::Configurable);
+}
+
+// UpdateLegacyRegExpStaticProperties ( C, S, startIndex, endIndex, capturedValues )
+// https://github.com/tc39/proposal-regexp-legacy-features#updatelegacyregexpstaticproperties--c-s-startindex-endindex-capturedvalues-
+void RegExpConstructor::update_legacy_properties(Utf16String const& string, size_t start_index, size_t end_index, Vector<Utf16View>& captured_values)
+{
+    // 1. Assert: C is an Object that has a [[RegExpInput]] internal slot.
+    // 2. Assert: Type(S) is String.
+
+    // 3. Let len be the number of code units in S.
+    size_t len = string.length_in_code_units();
+
+    // 4. Assert: startIndex and endIndex are integers such that 0 ≤ startIndex ≤ endIndex ≤ len.
+    VERIFY(start_index <= end_index);
+    VERIFY(end_index <= len);
+
+    // 5. Assert: capturedValues is a List of Strings.
+
+    // 6. Let n be the number of elements in capturedValues.
+    size_t group_count = captured_values.size();
+
+    // 7. Set the value of C’s [[RegExpInput]] internal slot to S.
+    m_legacy_properties.set_input(string);
+
+    // 8. Set the value of C’s [[RegExpLastMatch]] internal slot to a String whose length is endIndex - startIndex and containing the code units from S with indices startIndex through endIndex - 1, in ascending order.
+    auto const& lastMatch = Utf16String(string.view().substring_view(start_index, end_index - start_index));
+    m_legacy_properties.set_last_match(lastMatch);
+
+    // 9. If n > 0, set the value of C’s [[RegExpLastParen]] internal slot to the last element of capturedValues.
+    if (group_count > 0) {
+        auto& item = captured_values[group_count - 1];
+        m_legacy_properties.set_last_paren(Utf16String(item));
+    }
+    // 10. Else, set the value of C’s [[RegExpLastParen]] internal slot to the empty String.
+    else {
+        auto const& last_paren = Utf16String(String::empty().view());
+        m_legacy_properties.set_last_paren(last_paren);
+    }
+
+    // 11. Set the value of C’s [[RegExpLeftContext]] internal slot to a String whose length is startIndex and containing the code units from S with indices 0 through startIndex - 1, in ascending order.
+    auto const& left_context = Utf16String(string.view().substring_view(0, start_index));
+    m_legacy_properties.set_left_context(left_context);
+
+    // 12. Set the value of C’s [[RegExpRightContext]] internal slot to a String whose length is len - endIndex and containing the code units from S with indices endIndex through len - 1, in ascending order.
+    auto const& right_context = Utf16String(string.view().substring_view(end_index, len - end_index));
+    m_legacy_properties.set_right_context(right_context);
+
+    // 13. For each integer i such that 1 ≤ i ≤ 9
+    for (size_t i = 0; i < group_count; i++) {
```

Comment:
Is `group_count` guaranteed to be <= 9 at this point? If so, please `VERIFY()` it beforehand.